### DISCUSSION TITLE
Host IP (your-interface-ip) when running in Kubernetes

### DISCUSSION THREAD

**Discussion by kchaSCK** [thomaxxl/safrs#153] [Q&A]
Hi Thomas, all, 

this is a very cool and useful project. I'm trying to set it up in Kubernetes for a small prototype, but I'm a bit struggling with the host address (your-interface-ip) that needs to be passed. Both in Docker and K8s if I start one of your demos with "localhost" or "127.0.0.1", I can only access the API from within the container. It does not serve outside. In docker if I start it with the docker eth ip, for example 172.17.0.2, it serves fine, as long as docker assigns it that IP. 

Unfortunately in Kubernetes, pods have different IPs every time, can have multiple IPs if ran with multiple instances and overall having a static IP is an anti-pattern. 

Is there any way to open it up to serve on all interfaces / IPs that comes out of the box, or should I write some small function to figure out the pod IP address before really starting SAFRS? 

Thanks!

**Comment by thomaxxl** [comment]
Hi,

Did you try using "0.0.0.0" in flask `app.run(host='0.0.0.0')`? this will bind the app to all interfaces.

You probably also want to set the `swgger_host` argument to `SafrsApi()` to the (external) ip or domain name you will use.

**Reply by kchaSCK** [reply]
Yes, 0.0.0.0 works. Thanks Thomas!
